src/main/java/org/sonarlint/intellij/core/BackendService.kt

```diff
@@ -596,6 +600,24 @@ class BackendService @NonInjectable constructor(private val backend: SonarLintBa
     }
 
     companion object {
+        lateinit var backendLauncherJavaImpl: SonarLintRpcServerImpl;
+        fun initBackend(): SonarLintRpcServer {
+            val clientToServerOutputStream = PipedOutputStream()
+            val clientToServerInputStream = PipedInputStream(clientToServerOutputStream)
+
+            val serverToClientOutputStream = PipedOutputStream()
+            val serverToClientInputStream = PipedInputStream(serverToClientOutputStream)
+
+            var backendLauncher = BackendJsonRpcLauncher(clientToServerInputStream, serverToClientOutputStream)
```

Comment:
backendLauncher and clientLauncher need to be closed at the end, so you should probably keep them as a field somehow. See what I did in SLE as an example: https://github.com/SonarSource/sonarlint-eclipse/pull/563/files#diff-84d18f2ab5fb78caecbf75d8c25b8f43ca1812c0943fce805e8b504e71e2038eR248